### ISSUE TITLE
[Feature Request] suggest static anonymous functions

### ISSUE THREAD

**Issue by vladyslavstartsev** [kalessil/phpinspectionsea#1043]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Feature request |
| Plugin         | Php Inspections (EA Extended) or Php Inspections (EA Ultimate), any version |
| Language level | e.g. PHP 5.4  |

Current behaviour (description/screenshot:)
--
no behaviour
Expected behaviour
--
if anonymous function doesn't use `$this` suggest to make it static, for example
from this
```php
$this->app->bind($repository, function () use ($entity) {
    return \EntityManager::getRepository($entity);
});
```
to this
```php
$this->app->bind($repository, static function () use ($entity) {
    return \EntityManager::getRepository($entity);
});
```

**Comment by rentalhost** [comment]
Wow! I even know that it is possible. haha
It have some performance advantage?

**Comment by vladyslavstartsev** [comment]
@rentalhost propaply yes, have not tested yet :smile:

**Comment by rentalhost** [comment]
I did a local test, and it speed up ~13%, which make this feature a good catch for me. 😄

**Comment by kalessil** [comment]
Holy shit, how this possible at all. I mean since which PHP version?

**Comment by vladyslavstartsev** [comment]
@kalessil since php 5.4 see [docs](http://php.net/manual/en/functions.anonymous.php#functions.anonymous-functions.static)

**Comment by kalessil** [comment]
Thank you, @vladyslavstartsev. Will check docs for side-effects.

**Comment by kwvanderlinde** [comment]
Hello all! This is a neat idea, but keep in mind that static closures cannot be used in all the same situations that non-static closures can. E.g.:
```
$o = new \stdClass();
$c = static function () { };
$d = $c->bindTo($o);
```
This gives a warning: "Cannot bind an instance to a static closure", but this would be valid if `$c` were not marked `static`.

**Comment by kalessil** [comment]
Implemented!

@kwvanderlinde: confirmed, let's wait for first bug-reports and only then handle the false-positive.
